src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -355,23 +358,48 @@ public void GarbageCollect(Func<string, IWorkloadResolver> getResolverForWorkloa
 
             //  Garbage collect workload sets
             var installedWorkloadSets = _workloadResolver.GetWorkloadManifestProvider().GetAvailableWorkloadSets();
-            var manifestInstallDirForFeatureBand = GetManifestInstallDirForFeatureBand(_sdkFeatureBand.ToString());
-            string workloadSetsDirectory = Path.Combine(manifestInstallDirForFeatureBand, SdkDirectoryWorkloadManifestProvider.WorkloadSetsFolderName);
-
+            Dictionary<(string workloadSetVersion, SdkFeatureBand workloadSetFeatureBand), List<SdkFeatureBand>> workloadSetInstallRecords = GetAllWorkloadSetInstallRecords();
             foreach ((string workloadSetVersion, _) in installedWorkloadSets)
             {
-                if (garbageCollector.WorkloadSetsToKeep.Contains(workloadSetVersion))
+                WorkloadSet.WorkloadSetVersionToWorkloadSetPackageVersion(workloadSetVersion, out var workloadSetFeatureBand);
```

Comment:
This looks like you're trying to access the package version, which doesn't really make sense. Only afterwards did I realize that you really just want the feature band. Can't you just cut the incoming string at the third dot or first [+, -]?

Reply:
I much prefer to reuse existing logic instead of adding new ad-hoc parsing.  I've added a comment to help make it clearer.